lib/Controller/WopiController.php

```diff
@@ -485,10 +485,21 @@ public function putFile($fileId,
 
 			$content = fopen('php://input', 'rb');
 
+			// Get expected FileSize after writing
+			$storage = \OC_Helper::getStorageInfo($path);
+			fseek($content,0,SEEK_END);
+			$finalSize = ftell($content) + $file->getSize();
+			rewind( $content );
+
 			try {
-				$this->wrappedFilesystemOperation($wopi, function () use ($file, $content) {
-					return $file->putContent($content);
-				});
+				if ($finalSize <= $storage['free']){
+					$this->wrappedFilesystemOperation($wopi, function () use ($file, $content) {
+						return $file->putContent($content);
+					});
+				}
+				else {
+					throw new \Exception('Not enough storage');
+				}
```

Comment:
I personally find early return or throw a bit more readable:

```suggestion
				if ($finalSize > $storage['free']){
					throw new \Exception('Not enough storage');
				}
				
				$this->wrappedFilesystemOperation($wopi, function () use ($file, $content) {
					return $file->putContent($content);
				});
```